### DISCUSSION TITLE
Multiple instances

### DISCUSSION THREAD

**Discussion by Squiccio** [CS-SI/eodag#674] [Q&A]
I tried to launch 2 eodag instances at the same time but only one was ended succesfully. I thing it's related to the .config dir that contains temporary files generated during the run. How is it possible to execute multiple runs?

**Comment by sbrunato** [comment]
Hello @Squiccio, sharing the same home directory for multiple instances is not an expected scenario. What is your use-case for having 2 EODAG instances running at the same time?

**Reply by Squiccio** [reply]
Hi @sbrunato, since it is not possible to start multiple dowloads at same time (i didn't find info about this), i thought to split my query in more instances in order to have multiple downloads (one for each instance). I hope i explained well my idea

**Reply by sbrunato** [reply]
You can perform multiple downloads with a single `EODataAccessGateway` instance:

```py
import concurrent.futures
from eodag.utils import ProgressCallback

with concurrent.futures.ThreadPoolExecutor() as executor:
    futures = []
    paths = []
    for prod in search_results:
        futures.append(executor.submit(dag.download, prod, progress_callback=ProgressCallback(position=None)))
    for future in concurrent.futures.as_completed(futures):
        paths.append(future.result())
```

This may be simplified if we implement a new feature allowing passing an executor as argument of `download_all` (see https://github.com/CS-SI/eodag/issues/146)

**Reply by Squiccio** [reply]
Well i'm not so expert in programming but in my (simple) command line i think tht it's not possible to pass this instance. Actually the limit of parallels downloads is for me not easy to solve. Thanks anyway for your suppot
